dbt/macros/row_count_by_year.sql

```diff
@@ -0,0 +1,24 @@
+-- Macro that takes a model and returns its row count grouped and
+-- sorted by year.
+--
+-- The name of the column containing year data can be specified with
+-- the `year_column` argument, defaulting to "year". The sort order
+-- can be specified with the `ordering` argument, defaulting to "asc".
+--
+-- If the `print` argument is set to True (default is False), the macro will
+-- print the results of the query to stdout, which allows this macro to be used
+-- by scripts to return data.
+{% macro row_count_by_year(model, year_column="year", ordering="asc", print=False) %}
+    {% set query %}
+        select count(*) as COUNT, {{ year_column }}
+        from {{ model }}
+        group by {{ year_column }}
+        order by {{ year_column }} {{ ordering }}
+    {% endset %}
```

Comment:
**suggestion:** It might be worth just making this generalizable i.e. `row_count_by_group()`, since this is a super common pattern that we may want to re-use.

Reply:
Good idea, done in [64f056b](https://github.com/ccao-data/data-architecture/pull/156/commits/64f056bd3ec0933468aa59962dfebc997ce8e15f) 👍🏻